lib/sles4sap/sap_deployment_automation_framework/configure_tfvars.pm

```diff
@@ -116,34 +126,126 @@ sub set_workload_vnet_name {
-    my @variable_names = qw(SDAF_DB_IMAGE_PUBLISHER SDAF_DB_IMAGE_OFFER SDAF_DB_IMAGE_SKU SDAF_DB_IMAGE_VERSION);
+    my @variable_names = qw(SDAF_IMAGE_PUBLISHER SDAF_IMAGE_OFFER SDAF_IMAGE_SKU SDAF_IMAGE_VERSION);
     # This maps a variable name from array @variable names to value from delimited 'PUBLIC_CLOUD_IMAGE_ID' parameter
     # Order is important here
     @params{@variable_names} = split(':', get_required_var('PUBLIC_CLOUD_IMAGE_ID'));
 
     # Add all remaining parameters with static values
-    $params{SDAF_DB_IMAGE_OS_TYPE} = 'LINUX';    # this can be modified in case of non linux images
-    $params{SDAF_DB_SOURCE_IMAGE_ID} = '';    # for supplying uploaded image - not implemented yet
-    $params{SDAF_DB_IMAGE_TYPE} = 'marketplace';
+    $params{SDAF_IMAGE_OS_TYPE} = 'LINUX';    # this can be modified in case of non linux images
+    $params{SDAF_SOURCE_IMAGE_ID} = '';    # for supplying uploaded image - not implemented yet
+    $params{SDAF_IMAGE_TYPE} = 'marketplace';
 
     foreach (keys(%params)) {
         set_var($_, $params{$_});
     }
 }
+
+=head2 set_hana_db_parameters
+
+    set_hana_db_parameters(components=>['db_install', 'db_ha']);
+
+Sets tfvars Database HA parameters according to scenario defined by B<$args{components}>.
+
+=over
+
+=item * B<components>: B<ARRAYREF> of components that are to be installed. Check function B<validate_components> for available options.
+
+=back
+
+=cut
+
+sub set_hana_db_parameters {
+    my (%args) = @_;
+    # Enable HA cluster
+    int(grep /ha/, @{$args{components}})
+      ? set_var('SDAF_HANA_HA_SETUP', 'true') : set_var('SDAF_HANA_HA_SETUP', 'false');
+}
+
+=head2 set_netweaver_parameters
+
+    set_netweaver_parameters(components=>['db_install', 'db_ha']);
+
+Sets tfvars parameters related to SAP Netweaver according to scenario defined by B<$args{components}>.
+
+=over
+
+=item * B<components>: B<ARRAYREF> of components that are to be installed. Check function B<validate_components> for available options.
+
+=back
+
+=cut
+
+sub set_netweaver_parameters {
+    my (%args) = @_;
+    # Default values - everything turned off
+    my %parameters = (
+        SDAF_ASCS_SERVER => 0,
+        SDAF_APP_SERVER_COUNT => 0,
+        SDAF_ERS_SERVER => 'false'
+    );
+
+    # All nw_* scenarios require ASCS deployment
+    $parameters{SDAF_ASCS_SERVER} = 1 if grep /nw/, @{$args{components}};
+
+    # So far 1x PAS and 1x AAS should be enough for coverage
+    my $app_servers =
+      int(grep /pas/, @{$args{components}}) + int(grep /aas/, @{$args{components}});
+
+    $parameters{SDAF_APP_SERVER_COUNT} = $app_servers;
+    $parameters{SDAF_ERS_SERVER} = int(grep /ensa/, @{$args{components}}) ? 'true' : 'false';
+
+    for my $parameter (keys(%parameters)) {
+        set_var($parameter, $parameters{$parameter});
+    }
+}
+
+=head2 validate_components
+
+    validate_components(components=>['db_install', 'db_ha']);
```

Comment:
Is this function about checking a list that come from openQA variable `SDAF_DEPLOYMENT_SCENARIO` against an hard-coded list in perl that represent what sdaf or the osado test code is supporting?

Reply:
Yes exactly.
How to use the parameter is described in test module as well.